WordPress/Classes/ViewRelated/Reader/ReaderSearchViewController.swift

```diff
@@ -169,13 +173,35 @@ import Gridicons
     // MARK: - Configuration
 
 
-    @objc func setupSearchBar() {
+    private func setupSearchBar() {
         // Appearance must be set before the search bar is added to the view hierarchy.
         let placeholderText = NSLocalizedString("Search WordPress", comment: "Placeholder text for the Reader search feature.")
         UITextField.appearance(whenContainedInInstancesOf: [UISearchBar.self, ReaderSearchViewController.self]).placeholder = placeholderText
 
         searchBar.becomeFirstResponder()
         WPStyleGuide.configureSearchBar(searchBar)
+        guard AppConfiguration.isWordPress, FeatureFlag.jetpackPowered.enabled else {
+            return
+        }
+        searchBar.inputAccessoryView = bannerView
+        hideBannerViewIfNeeded()
+    }
+
+    /// hides the Jetpack powered banner on iPhone landscape
+    private func hideBannerViewIfNeeded() {
+        guard AppConfiguration.isWordPress, FeatureFlag.jetpackPowered.enabled else {
+            return
+        }
+        let height = UIApplication.shared.mainWindow?.frame.size.height ?? 0
+        // maximum height of any iPhone in landscape. iPads and portrait orientations
+        // all have heights greater than this one.
+        let maximumLandscapeHeight: CGFloat = 428
+        bannerView.isHidden = height <= maximumLandscapeHeight
```

Comment:
Perhaps a cleaner solution would be to use size classes here?

For example something like this:
```
if traitCollection.verticalSizeClass == .compact {
    bannerView.isHidden = true
} else {
    bannerView.isHidden = false
}
```

Reply:
Indeed we should use the size class 🤦 , thanks for bringing this up @sla8c !